f3/api.go

```diff
@@ -51,11 +51,19 @@ type Signer interface {
 	Verify(sender ActorID, msg, sig []byte) bool
 }
 
+type Aggregator interface {
+	// Aggregates signatures from a participant to an existing signature.
+	Aggregate(sig []byte, aggSignature []byte) []byte
+	// VerifyAggregate verifies an aggregate signature.
+	VerifyAggregate(payload, aggSig []byte, signers map[ActorID]struct{}) bool
```

Comment:
This API seems to assume that the implementor knows how to get a public key for an ActorID somewhere. It works for the fake impl, but I don't think will for the real one. I think instead `signers` should just be a list of public keys, so that it becomes a pure function.

Reply:
Done.

Reply:
Actually, I think it would be most natural for aggregate signature itself to encode the list of signers (e.g., as a bitmask with respect to the publicly known power table).

Reply:
f3 will need to know about public keys either way, as it needs to store them in the power table/state. Yes, we could abstract pubkeys out in this interface, but then it means that the implementer of the API has to have access to the context-sensitive (per granite instance) power table

Reply:
Implemented the verification of the proper public keys as listed [suggested in a previous PR](https://github.com/filecoin-project/go-f3/pull/32#discussion_r1453955206)